src/plugins/content_management/public/types.ts

```diff
@@ -0,0 +1,28 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import React from 'react';
+import { CoreStart } from 'opensearch-dashboards/public';
+
+import { ContentManagementService, ContentProvider } from './services';
+import { EmbeddableSetup, EmbeddableStart } from '../../embeddable/public';
+
+export interface ContentManagementPluginSetup {
+  registerPage: ContentManagementService['registerPage'];
+}
+export interface ContentManagementPluginStart {
+  registerContentProvider: (provider: ContentProvider) => void;
```

Comment:
At setup phase, the page and the sections will be registered, so at start phase, when register contents, the page and sections can be found. But essential, we could make it so that when registering contents, it doesn't rely on the existence of page and sections. I'd leave as it is util there is a need to register content at setup phase.